Modules/_sqlite/connection.c

```diff
@@ -449,7 +451,9 @@ connection_close(pysqlite_Connection *self)
             if (_Py_IsInterpreterFinalizing(PyInterpreterState_Get())) {
                 remove_callbacks(self->db);
             }
-            (void)connection_exec_stmt(self, "ROLLBACK");
+            if (connection_exec_stmt(self, "ROLLBACK") < 0) {
+                PyErr_WriteUnraisable(self);
```

Comment:
Hm, if part of `tp_finalize` we'd want to write an unraisable exception, but for normal `.close()`, we'd want to raise an ordinary exception.

Reply:
You have to cast the argument to ``PyObject*`` to avoid compiler warnings.

Reply:
Yeah, I noticed; I just had to fix a compiler error in my other PR first ;)

Reply:
Be careful, PyErr_WriteUnraisable() can resurect an object if sys.unraisablehook is overriden and the new hook keeps the object alive: https://docs.python.org/dev/library/sys.html#sys.unraisablehook

Calling it from dealloc is dangerous.

Reply:
> Be careful, PyErr_WriteUnraisable() can resurect an object if sys.unraisablehook is overriden and the new hook keeps the object alive: https://docs.python.org/dev/library/sys.html#sys.unraisablehook
> 
> Calling it from dealloc is dangerous.

I think we should emphasise that warning in the docs, perhaps using `.. warning::` in Sphinx. IMO there is not much we can do about that. The alternative is to clear the exception.

Reply:
I created gh-108105 in order to better emphasise the dangers of the unraisable hook, and to clean up some markups.